addons/mass_mailing/models/mailing_contact.py

```diff
@@ -176,6 +194,16 @@ def action_add_to_mailing_list(self):
 
         return action
 
+    def action_view_mailings(self):
```

Comment:
Basically the same comment as the statistics, you can't do that because the database is not able to efficiently filter mailing traces based on "model" and "res_id".

I understand these buttons are available from the mailing.mailing view but that is a different story as it uses the existing "mass_mailing_id" to filter records which is fully indexed.